pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -0,0 +1,792 @@
+			if nodeSelector := state.availableOnNodes[index]; nodeSelector != nil {
+				if !nodeSelector.Match(node) {
+					logger.V(5).Info("AvailableOnNodes does not match", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclaim", klog.KObj(claim))
+					unavailableClaims = append(unavailableClaims, index)
+				}
+			}
+		case claim.Status.DeallocationRequested:
+			// We shouldn't get here. PreFilter already checked this.
+			return statusUnschedulable(logger, "resourceclaim must be reallocated", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclaim", klog.KObj(claim))
+		case claim.Spec.AllocationMode == resourcev1alpha1.AllocationModeWaitForFirstConsumer:
+			// The ResourceClass might have a node filter. This is
+			// useful for trimming the initial set of potential
+			// nodes before we ask the driver(s) for information
+			// about the specific pod.
+			class, err := pl.classLister.Get(claim.Spec.ResourceClassName)
+			if err != nil {
+				// If the class does not exist, then allocation cannot proceed.
+				return statusError(logger, fmt.Errorf("look up resource class: %v", err))
+			}
+			if class.SuitableNodes != nil {
+				// TODO (#113700): parse class.SuitableNodes once in PreFilter, reuse result.
+				matches, err := corev1helpers.MatchNodeSelectorTerms(node, class.SuitableNodes)
+				if err != nil {
+					return statusError(logger, fmt.Errorf("potential node filter: %v", err))
+				}
+				if !matches {
+					return statusUnschedulable(logger, "excluded by resource class node filter", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclass", klog.KObj(class))
+				}
+			}
+
+			// Now we need information from drivers.
+			podScheduling, err := state.initializePodScheduling(ctx, pod, pl.podSchedulingLister)
+			if err != nil {
+				return statusError(logger, err)
+			}
+			status := statusForClaim(podScheduling, pod.Spec.ResourceClaims[index].Name)
+			if status != nil {
+				for _, unsuitableNode := range status.UnsuitableNodes {
+					if node.Name == unsuitableNode {
+						return statusUnschedulable(logger, "resourceclaim cannot be allocated for the node (unsuitable)", "pod", klog.KObj(pod), "node", klog.KObj(node), "resourceclaim", klog.KObj(claim), "unsuitablenodes", status.UnsuitableNodes)
+					}
+				}
+			}
+		default:
+			// This should have been delayed allocation. Immediate
+			// allocation was already checked for in PreFilter.
+			return statusError(logger, fmt.Errorf("internal error, unexpected allocation mode %v", claim.Spec.AllocationMode))
+		}
+	}
+
+	if len(unavailableClaims) > 0 {
+		state.mutex.Lock()
+		defer state.mutex.Unlock()
+		if state.unavailableClaims == nil {
+			state.unavailableClaims = sets.NewInt()
+		}
+		state.unavailableClaims.Insert(unavailableClaims...)
+		return statusUnschedulable(logger, "resourceclaim not available on the node", "pod", klog.KObj(pod))
+	}
+
+	return nil
+}
+
+// PostFilter checks whether freeing an allocated claim might help to get a Pod
```

Comment:
I think we're not checking here, just reclaim the resources.

Reply:
There are still some checks that are done by the function itself (are there claims? can they be deallocated).

Let me rephrase:
```
// PostFilter checks whether there are allocated claims that
// could get deallocated to help get the Pod schedulable.
// If yes, it picks one and requests its deallocation.
// This only gets called when filtering found no suitable node.
```